### PR TITLE
cleanup macros

### PR DESCRIPTION
**Pull request by OmranAbdelkafi**
## Description 
cleanup the macros under ebDefinitions.h 

## Performance
Lossless

### CODE REVIEW

**Comment by 1480c1** [comment]
```c
/home/travis/build/OpenVisualCloud/SVT-AV1/Source/Lib/Encoder/Codec/EbEncDecProcess.c: In function ‘signal_derivation_enc_dec_kernel_oq’:
/home/travis/build/OpenVisualCloud/SVT-AV1/Source/Lib/Encoder/Codec/EbEncDecProcess.c:1800:10: error: this ‘if’ clause does not guard... [-Werror=misleading-indentation]
     else if (context_ptr->pd_pass == PD_PASS_1)
          ^~
/home/travis/build/OpenVisualCloud/SVT-AV1/Source/Lib/Encoder/Codec/EbEncDecProcess.c:1802:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
         if (pcs_ptr->enc_mode <= ENC_M1 ||
         ^~
cc1: all warnings being treated as errors
```

**Comment by tianjunwork** [comment]
One more error to address
```
/home/travis/build/OpenVisualCloud/SVT-AV1/Source/Lib/Encoder/Codec/EbMotionEstimationProcess.c:259:9: error: misleading indentation; statement is not part of the previous 'else' [-Werror,-Wmisleading-indentation]
        context_ptr->me_context_ptr->hme_search_method = SUB_SAD_SEARCH;
        ^
/home/travis/build/OpenVisualCloud/SVT-AV1/Source/Lib/Encoder/Codec/EbMotionEstimationProcess.c:256:7: note: previous statement is here
    } else
      ^
```

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 1]
```diff
@@ -316,77 +303,72 @@ void inter_intra_search(PictureControlSet *pcs_ptr, ModeDecisionContext *context
...
+            pcs_ptr,
+            pcs_ptr->parent_pcs_ptr->enhanced_picture_ptr,
+            (EbReferenceObject *)pcs_ptr->ref_pic_ptr_array[list_idx0][list_idx0]->object_ptr,
+            &ref_pic_list0);
+    if (ref_pic_list1 != NULL)
+        use_scaled_rec_refs_if_needed(
+            pcs_ptr,
+            pcs_ptr->parent_pcs_ptr->enhanced_picture_ptr,
+            (EbReferenceObject *)pcs_ptr->ref_pic_ptr_array[list_idx1][ref_idx_l1]->object_ptr,
+            &ref_pic_list1);
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:336]: Is the style change here necessary? seems not consistent with below.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 2]
```diff
@@ -897,59 +877,34 @@ void unipred_3x3_candidates_injection(const SequenceControlSet *scs_ptr, Picture
...
+                        continue;
                 }
-#if ENHANCED_ME_MV
                 int16_t to_inject_mv_x;
                 int16_t to_inject_mv_y;
                 if (pcs_ptr->parent_pcs_ptr->frm_hdr.allow_high_precision_mv) {
                     to_inject_mv_x = context_ptr->sb_me_mv[context_ptr->blk_geom->blkidx_mds]
-                        [REF_LIST_0][list0_ref_index][0] +
+                                                          [REF_LIST_0][list0_ref_index][0] +
                         bipred_3x3_x_pos[bipred_index];
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:896]: style here is messed up.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 3]
```diff
@@ -1064,73 +1019,38 @@ void unipred_3x3_candidates_injection(const SequenceControlSet *scs_ptr, Picture
...
+                            continue;
                     }
-#if ENHANCED_ME_MV
                     int16_t to_inject_mv_x;
                     int16_t to_inject_mv_y;
                     if (pcs_ptr->parent_pcs_ptr->frm_hdr.allow_high_precision_mv) {
                         to_inject_mv_x = context_ptr->sb_me_mv[context_ptr->blk_geom->blkidx_mds]
-                            [REF_LIST_1][list1_ref_index][0] +
+                                                              [REF_LIST_1][list1_ref_index][0] +
                             bipred_3x3_x_pos[bipred_index];
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1039]: style here is messed up.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 4]
```diff
@@ -1259,12 +1179,12 @@ void bipred_3x3_candidates_injection(const SequenceControlSet *scs_ptr, PictureC
     BlockSize    bsize          = context_ptr->blk_geom->bsize;
     MD_COMP_TYPE tot_comp_types = (pcs_ptr->parent_pcs_ptr->compound_mode == 1 ||
                                    context_ptr->compound_types_to_try == MD_COMP_AVG)
-                                      ? MD_COMP_AVG
-                                      : (bsize >= BLOCK_8X8 && bsize <= BLOCK_32X32)
-                                            ? context_ptr->compound_types_to_try
-                                            : context_ptr->compound_types_to_try == MD_COMP_WEDGE
-                                                  ? MD_COMP_DIFF0
-                                                  : context_ptr->compound_types_to_try;
+        ? MD_COMP_AVG
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1182]: style change here is not necessary

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 5]
```diff
@@ -1288,100 +1208,42 @@ void bipred_3x3_candidates_injection(const SequenceControlSet *scs_ptr, PictureC
...
                     int16_t to_inject_mv_x_l1;
                     int16_t to_inject_mv_y_l1;
                     if (pcs_ptr->parent_pcs_ptr->frm_hdr.allow_high_precision_mv) {
-                        to_inject_mv_x_l1 =
-                            context_ptr
-                            ->sb_me_mv[context_ptr->blk_geom->blkidx_mds]
-                            [me_block_results_ptr->ref1_list][list1_ref_index][0] +
+                        to_inject_mv_x_l1 = context_ptr->sb_me_mv[context_ptr->blk_geom->blkidx_mds]
+                                                                 [me_block_results_ptr->ref1_list]
+                                                                 [list1_ref_index][0] +
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1226]: style here is messed up

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 6]
```diff
@@ -1394,12 +1256,12 @@ void bipred_3x3_candidates_injection(const SequenceControlSet *scs_ptr, PictureC
                                                               mi_col,
                                                               mi_row,
                                                               context_ptr->blk_geom->bsize) &&
-                                      is_inside_tile_boundary(&(xd->tile),
-                                                              to_inject_mv_x_l1,
-                                                              to_inject_mv_y_l1,
-                                                              mi_col,
-                                                              mi_row,
-                                                              context_ptr->blk_geom->bsize);
+                            is_inside_tile_boundary(&(xd->tile),
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1259]: the previous style makes more sense.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 7]
```diff
@@ -1497,92 +1359,42 @@ void bipred_3x3_candidates_injection(const SequenceControlSet *scs_ptr, PictureC
...
                     int16_t to_inject_mv_x_l0;
                     int16_t to_inject_mv_y_l0;
                     if (pcs_ptr->parent_pcs_ptr->frm_hdr.allow_high_precision_mv) {
-                        to_inject_mv_x_l0 =
-                            context_ptr
-                            ->sb_me_mv[context_ptr->blk_geom->blkidx_mds]
-                            [me_block_results_ptr->ref0_list][list0_ref_index][0] +
+                        to_inject_mv_x_l0 = context_ptr->sb_me_mv[context_ptr->blk_geom->blkidx_mds]
+                                                                 [me_block_results_ptr->ref0_list]
+                                                                 [list0_ref_index][0] +
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1371]: style here is messed up.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 8]
```diff
@@ -1595,12 +1407,12 @@ void bipred_3x3_candidates_injection(const SequenceControlSet *scs_ptr, PictureC
                                                               mi_col,
                                                               mi_row,
                                                               context_ptr->blk_geom->bsize) &&
-                                      is_inside_tile_boundary(&(xd->tile),
-                                                              to_inject_mv_x_l1,
-                                                              to_inject_mv_y_l1,
-                                                              mi_col,
-                                                              mi_row,
-                                                              context_ptr->blk_geom->bsize);
+                            is_inside_tile_boundary(&(xd->tile),
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1410]: previous style makes more sense.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 9]
```diff
@@ -1800,11 +1610,11 @@ void inject_mvp_candidates_ii(struct ModeDecisionContext *context_ptr, PictureCo
...
             uint8_t inter_type;
-            uint8_t is_ii_allowed =
-                svt_is_interintra_allowed(context_ptr->md_enable_inter_intra, bsize, NEARESTMV, rf);
+            uint8_t is_ii_allowed = svt_is_interintra_allowed(
+                context_ptr->md_enable_inter_intra, bsize, NEARESTMV, rf);
             uint8_t tot_inter_types = is_ii_allowed ? II_COUNT : 1;
-            uint8_t is_obmc_allowed =
-                obmc_motion_mode_allowed(pcs_ptr, context_ptr, bsize, rf[0], rf[1], NEARESTMV) ==
+            uint8_t is_obmc_allowed = obmc_motion_mode_allowed(
+                                          pcs_ptr, context_ptr, bsize, rf[0], rf[1], NEARESTMV) ==
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1617]: style here is messed up.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 10]
```diff
@@ -2016,19 +1824,19 @@ void inject_mvp_candidates_ii(struct ModeDecisionContext *context_ptr, PictureCo
...
-                                                      to_inject_mv_y_l1,
-                                                      mi_col,
-                                                      mi_row,
-                                                      context_ptr->blk_geom->bsize);
+                    is_inside_tile_boundary(&(xd->tile),
+                                            to_inject_mv_x_l1,
+                                            to_inject_mv_y_l1,
+                                            mi_col,
+                                            mi_row,
+                                            context_ptr->blk_geom->bsize);
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1832]: previous style makes more sense.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 11]
```diff
@@ -2121,19 +1929,19 @@ void inject_mvp_candidates_ii(struct ModeDecisionContext *context_ptr, PictureCo
...
-                                                          to_inject_mv_y_l1,
-                                                          mi_col,
-                                                          mi_row,
-                                                          context_ptr->blk_geom->bsize);
+                        is_inside_tile_boundary(&(xd->tile),
+                                                to_inject_mv_x_l1,
+                                                to_inject_mv_y_l1,
+                                                mi_col,
+                                                mi_row,
+                                                context_ptr->blk_geom->bsize);
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:1937]: also here.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 12]
```diff
@@ -2285,22 +2073,21 @@ void inject_new_nearest_new_comb_candidates(const SequenceControlSet *  scs_ptr,
...
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:2081]: also here.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 13]
```diff
@@ -2407,22 +2185,20 @@ void inject_new_nearest_new_comb_candidates(const SequenceControlSet *  scs_ptr,
...
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:2193]: also here.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 14]
```diff
@@ -3581,12 +3092,12 @@ void inject_new_candidates(const SequenceControlSet *  scs_ptr,
...
-                                            ? context_ptr->compound_types_to_try
-                                            : context_ptr->compound_types_to_try == MD_COMP_WEDGE
-                                                  ? MD_COMP_DIFF0
-                                                  : context_ptr->compound_types_to_try;
+        ? MD_COMP_AVG
+        : (bsize >= BLOCK_8X8 && bsize <= BLOCK_32X32)
+            ? context_ptr->compound_types_to_try
+            : context_ptr->compound_types_to_try == MD_COMP_WEDGE
+                ? MD_COMP_DIFF0
+                : context_ptr->compound_types_to_try;
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:3100]: previous style makes more sense?

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 15]
```diff
@@ -3637,15 +3140,15 @@ void inject_new_candidates(const SequenceControlSet *  scs_ptr,
...
-                uint8_t is_ii_allowed =
-                    svt_is_interintra_allowed(context_ptr->md_enable_inter_intra, bsize, NEWMV, rf);
+                uint8_t is_ii_allowed = svt_is_interintra_allowed(
+                    context_ptr->md_enable_inter_intra, bsize, NEWMV, rf);
                 uint8_t tot_inter_types = is_ii_allowed ? II_COUNT : 1;
-                uint8_t is_obmc_allowed =
-                    obmc_motion_mode_allowed(pcs_ptr, context_ptr, bsize, rf[0], rf[1], NEWMV) ==
+                uint8_t is_obmc_allowed = obmc_motion_mode_allowed(
+                                              pcs_ptr, context_ptr, bsize, rf[0], rf[1], NEWMV) ==
                     OBMC_CAUSAL;
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:3148]: style here is messed up

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 16]
```diff
@@ -4382,28 +3848,27 @@ void inject_inter_candidates(PictureControlSet *pcs_ptr, ModeDecisionContext *co
...
-                                            ? context_ptr->compound_types_to_try
-                                            : context_ptr->compound_types_to_try == MD_COMP_WEDGE
-                                                  ? MD_COMP_DIFF0
-                                                  : context_ptr->compound_types_to_try;
+        ? MD_COMP_AVG
+        : (bsize >= BLOCK_8X8 && bsize <= BLOCK_32X32)
+            ? context_ptr->compound_types_to_try
+            : context_ptr->compound_types_to_try == MD_COMP_WEDGE
+                ? MD_COMP_DIFF0
+                : context_ptr->compound_types_to_try;
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:3871]: previous style makes more sense?

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 17]
```diff
@@ -4382,28 +3848,27 @@ void inject_inter_candidates(PictureControlSet *pcs_ptr, ModeDecisionContext *co
...
-                                                                                    : EB_TRUE;
+    EbBool       is_compound_enabled      = (frm_hdr->reference_mode == SINGLE_REFERENCE) ? 0 : 1;
+    uint8_t      inj_mv                   = 1;
+    int          inside_tile              = 1;
+    MacroBlockD *xd                       = context_ptr->blk_ptr->av1xd;
+    int          umv0tile     = (scs_ptr->static_config.unrestricted_motion_vector == 0);
+    MeSbResults *me_results   = pcs_ptr->parent_pcs_ptr->me_results[context_ptr->me_sb_addr];
+    EbBool       allow_bipred = (context_ptr->blk_geom->bwidth == 4 ||
+                           context_ptr->blk_geom->bheight == 4)
+        ? EB_FALSE
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:3859]: need to fix style of the last two lines.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 18]
```diff
@@ -4434,11 +3902,11 @@ void inject_inter_candidates(PictureControlSet *pcs_ptr, ModeDecisionContext *co
...
-        EbBool allow_compound =
-            (frm_hdr->reference_mode == SINGLE_REFERENCE || context_ptr->blk_geom->bwidth == 4 ||
-             context_ptr->blk_geom->bheight == 4)
-                ? EB_FALSE
-                : EB_TRUE;
+        EbBool allow_compound = (frm_hdr->reference_mode == SINGLE_REFERENCE ||
+                                 context_ptr->blk_geom->bwidth == 4 ||
+                                 context_ptr->blk_geom->bheight == 4)
+            ? EB_FALSE
+            : EB_TRUE;
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:3909]: need to fix style of the last two lines.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 19]
```diff
@@ -5027,11 +4441,12 @@ void inject_intra_candidates_ois(PictureControlSet *pcs_ptr, ModeDecisionContext
     uint8_t                intra_mode;
     uint32_t               can_total_cnt   = 0;
     ModeDecisionCandidate *candidate_array = context_ptr->fast_candidate_array;
-    EbBool                 disable_cfl_flag =
-        (MAX(context_ptr->blk_geom->bheight, context_ptr->blk_geom->bwidth) > 32) ? EB_TRUE
-                                                                                  : EB_FALSE;
+    EbBool disable_cfl_flag = (MAX(context_ptr->blk_geom->bheight, context_ptr->blk_geom->bwidth) >
+                               32)
+        ? EB_TRUE
+        : EB_FALSE;
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:4447]: need to fix style of the last two lines.

File: Source/Lib/Encoder/Codec/EbModeDecision.c [hunk 20]
```diff
@@ -5169,25 +4582,26 @@ void eb_av1_init_me_luts(void) {
...
+    uint32_t        full_lambda = context_ptr->hbd_mode_decision
+        ? context_ptr->full_lambda_md[EB_10_BIT_MD]
+        : context_ptr->full_lambda_md[EB_8_BIT_MD];
     //fill x with what needed.
-    x->is_exhaustive_allowed =
-        context_ptr->blk_geom->bwidth == 4 || context_ptr->blk_geom->bheight == 4 ? 1 : 0;
+    x->is_exhaustive_allowed = context_ptr->blk_geom->bwidth == 4 ||
+            context_ptr->blk_geom->bheight == 4
+        ? 1
+        : 0;
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbModeDecision.c:4604]: need to fix style of the last two lines.

File: Source/Lib/Encoder/Codec/EbPictureDecisionProcess.c [hunk 21]
```diff
@@ -1221,34 +1147,26 @@ EbErrorType signal_derivation_multi_processes_oq(
...
-#endif
-        //Exit TX size search when all coefficients are zero
-        // 0: OFF
-        // 1: ON
-        pcs_ptr->tx_size_early_exit = 1;
-#if MUS_ME
+   //Exit TX size search when all coefficients are zero
+   // 0: OFF
+   // 1: ON
+    pcs_ptr->tx_size_early_exit = 1;
```
Comment (tianjunwork) [inline Source/Lib/Encoder/Codec/EbPictureDecisionProcess.c:1162]: unnecessary style change.

**Comment by tianjunwork** [comment]
Hi @OmranAbdelkafi , Source/Lib/Encoder/Codec/EbPictureDecisionProcess.c also needs to revert style change.

**Pull request merged**